ImageGallery/ImageGallery/ImageStore 2.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  DownloadStore.swift
+//  ImagaeGallery
+//
+//  Created by Nick Ryan on 6/18/24.
+//
+
+import SwiftUI
+import Combine
+import Foundation
+
+@MainActor
+class ImageStore: NSObject, ObservableObject, URLSessionDownloadDelegate {
```

Comment:
Not sure you need to implement `URLSessionDownloadDelegate` in this homework. You can just pass a ProgressView to the placeholder if the image is not downloaded yet. But if you need to show a percentage progress, then you'd need this.